src/hd.rs

```diff
@@ -181,111 +176,59 @@ impl HDWallet {
         }
     }
 
-    pub fn from_bare_xpub(
-        xpub: XyzPubKey,
-        network: Network,
-        gap_limit: u32,
-        initial_import_size: u32,
-        rescan_policy: RescanSince,
-    ) -> Result<Self> {
-        ensure!(
-            xpub.matches_network(network),
-            "xpub network mismatch, {} is {} and not {}",
-            xpub,
-            xpub.network,
-            network
-        );
-        Ok(Self::new(
-            xpub.extended_pubkey,
-            network,
-            xpub.script_type,
-            gap_limit,
-            initial_import_size,
-            rescan_policy,
-        ))
-    }
-
-    pub fn from_xpub(
-        xpub: XyzPubKey,
-        network: Network,
-        gap_limit: u32,
-        initial_import_size: u32,
-        rescan_policy: RescanSince,
-    ) -> Result<Vec<Self>> {
-        ensure!(
-            xpub.matches_network(network),
-            "xpub network mismatch, {} is {} and not {}",
-            xpub,
-            xpub.network,
-            network
-        );
-        Ok(vec![
-            // external chain (receive)
-            Self::new(
-                xpub.extended_pubkey
-                    .derive_pub(&*EC, &[ChildNumber::from(0)])?,
-                network,
-                xpub.script_type,
-                gap_limit,
-                initial_import_size,
-                rescan_policy,
-            ),
-            // internal chain (change)
-            Self::new(
-                xpub.extended_pubkey
-                    .derive_pub(&*EC, &[ChildNumber::from(1)])?,
-                network,
-                xpub.script_type,
-                gap_limit,
-                initial_import_size,
-                rescan_policy,
-            ),
-        ])
-    }
-
-    pub fn from_xpubs(
-        xpubs: &[(XyzPubKey, RescanSince)],
-        bare_xpubs: &[(XyzPubKey, RescanSince)],
+    pub fn from_descriptors(
```

Comment:
This should be renamed to something like `from_config()` and accept `descriptors`, `xpubs` and `bare_xpubs`. But this can also be left for later.